### PR TITLE
Offline sync failed notification

### PR DESCRIPTION
**Pull request by vargaat**
### What's new?
- Added offline sync failed notification.
- Fixed a memory leak in background activity requester.
- Disabled sync selector screen's Sync button in offline mode.

refs: MBL-17123
affects: Student
release note: none

test plan:
- Make sync fail by either putting a long sync to the background or by disabling wifi.
- As long as you're not on the sync progress screen you should receive a local notification about the failure.

## Checklist

- [ ] Follow-up e2e test ticket created
- [ ] A11y checked
- [ ] Tested on phone
- [ ] Tested on tablet
- [ ] Tested in dark mode
- [ ] Tested in light mode
- [ ] Approve from product

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This pull request will not generate a release note.</td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-17123](https://instructure.atlassian.net/browse/MBL-17123)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 92.39% | 92.38% | 0.01%
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4688c1a1e9f51f0e4c28065659f1abfb44d4e7fe
</p>

File: Core/Core/CourseSync/CourseSyncDownloader/CourseSyncDownloaderAssembly.swift [hunk 1]
```diff
@@ -42,7 +42,7 @@ public enum CourseSyncDownloaderAssembly {
         return CourseSyncInteractorLive(contentInteractors: contentInteractors,
                                         filesInteractor: CourseSyncFilesInteractorLive(),
                                         progressWriterInteractor: CourseSyncProgressWriterInteractorLive(),
-                                        successNotification: CourseSyncSuccessNotificationInteractor(notificationManager: .shared,
+                                        notification: CourseSyncNotificationInteractor(notificationManager: .shared,
```
Comment (szabinst) [inline Core/Core/CourseSync/CourseSyncDownloader/CourseSyncDownloaderAssembly.swift:45]: I'd probably rename this to `notificationInteractor` so it's more consistent with the other property namings.

**Review by szabinst — APPROVED** [review]
QA + 1

**Review by balintbartok — APPROVED** [review]
QA + 1

File: Core/Core/CourseSync/CourseSyncDownloader/CourseSyncDownloaderAssembly.swift (hunk 1 above)
Comment (vargaat) [reply Core/Core/CourseSync/CourseSyncDownloader/CourseSyncDownloaderAssembly.swift]: Agree, renamed.

**Pull request merged**